apps/backup_and_restore/backup_and_restore.py

```diff
@@ -0,0 +1,429 @@
+import weaviate
+import datetime
+import time
+from loguru import logger
+from typing import Optional
+import random
+import numpy as np
+import uuid
+import sys
+import requests
+
+def other_classes(all_classes, self):
+    return [c for c in all_classes if c != self]
+
+def reset_schema(client: weaviate.Client, class_names):
+    client.schema.delete_all()
+    for class_name in class_names:
+        class_obj = {
+            "vectorizer": "none",
+            "vectorIndexConfig":{
+                "efConstruction": 128,
+                "maxConnections": 16,
+                "ef": 256,
+                "cleanupIntervalSeconds": 10,
+            },
+            "class": class_name,
+            "invertedIndexConfig":{
+                "indexTimestamps":False,
+            },
+            "properties": [
+                {
+                    "dataType": [ "boolean" ],
+                    "name": "should_be_deleted",
+                },
+                {
+                    "dataType": [ "boolean" ],
+                    "name": "is_divisible_by_four",
+                },
+                {
+                    "dataType": [ "int" ],
+                    "name": "index_id",
+                },
+                {
+                    "dataType": [ "string" ],
+                    "name": "stage",
+                },
+            ]
+        }
+
+        client.schema.create_class(class_obj)
+
+    for class_name in class_names:
+        for other in other_classes(class_names, class_name):
+            add_prop = {
+              "dataType": [
+                  other,
+              ],
+              "name": f"to_{other}"
+            }
+
+            client.schema.property.create(class_name, add_prop)
+
+def handle_errors(results: Optional[dict]) -> None:
+    """
+    Handle error message from batch requests logs the message as an info message.
+    Parameters
+    ----------
+    results : Optional[dict]
+        The returned results for Batch creation.
+    """
+
+    if results is not None:
+        for result in results:
+            if (
+                'result' in result
+                and 'errors' in result['result']
+                and 'error' in result['result']['errors']
+            ):
+                for message in result['result']['errors']['error']:
+                    logger.error(message['message'])
+
+def load_records(client: weaviate.Client, class_name="Class", start=0, end=100_000, stage="stage_0", all_classes=[]):
```

Comment:
Doing  `all_classes=[]` as a default parameter is really dangerous in python. The list is created once at "compile time" and then reused on every function call. I don't think it is a problem here as you always call it with an explicit argument, but that should be avoided unless you really want this behavior.

you can do 
```  
def load_records(client: weaviate.Client, class_name="Class", start=0, end=100_000, stage="stage_0", all_classes:Optional[List[str]]=None):
  if all_classes is None:
      all_classes=[]
```

You also seem to never use the default parameters, so maybe just remove them? If you want typing, you can do `var_name:type`

Reply:
Thanks, I had no idea. I'll update this!